tests/PHPStan/Command/IgnoredRegexValidatorTest.php

```diff
@@ -104,13 +104,15 @@ public function dataValidate(): array
 				'~Result of || is always true.~',
 				[],
 				false,
-				true,
+				false,
```

Comment:
This is a blocker, I can't merge this. This is the main reason why IgnoredRegexValidator exists at all.

Reply:
I do not understand what the test does. Please tell me what to do or feel free to push directly.

Reply:
IgnoredRegexValidator makes sure that if people have this in their configuration:

```
parameters:
	ignoreErrors:
		- '~Result of || is always true.~'

```

It makes PHPStan fail with the following error:

```
Invalid entry in ignoreErrors:
Ignored error ~Result of || is always true.~ has an unescaped '||' which leads to ignoring all errors. Use '\|\|' instead.
```

Reply:
Sure, but it MUST NOT fail, as it is valid regex :)

So please tell me how to make everyone happy.

Reply:
It's not a valid regex for `ignoreErrors`.

Reply:
Why?

(I know it is probably a mistate as unescaped, but in general, all regexes are supported)

Reply:
@ondrejmirtes I have updated the PR so these "ignoreErrors errors" are reported as before.

Reply:
@ondrejmirtes these "obviously unescaped `ignoreErrors` cases" are not handled. Can you check?

Reply:
@ondrejmirtes can you please check and let me know if there is anything to address?

Reply:
@ondrejmirtes would you mind sharing your review?

Reply:
Maybe in December, after I release 2.0. Don't expect it sooner.

Reply:
This is a bugfix into 1.12.x, might I understand why so late?

Reply:
Because I have different priorities and this is not a severe bug affecting a lot of users.

Reply:
@ondrejmirtes is now a better time?

About the usecase, see https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.65.0/src/DocBlock/TypeExpression.php#L62 for example.

`(a|b|)` is shorter syntax of `((?:a|b|)?)`, the first one is currently now parsed by PHPStan and this PR fixes it.

Reply:
@ondrejmirtes the https://github.com/phpstan/phpstan-src/pull/3507#discussion_r1789111901 review note has been addressed